samples/matmul/matmul_basic.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright 2025 Advanced Micro Devices, Inc.
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <fusilli.h>
+
+#include "utils.h"
+
+#include <catch2/catch_test_macros.hpp>
+
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <tuple>
+#include <unordered_map>
+#include <vector>
+
+using namespace fusilli;
+
+TEST_CASE("Matrix multiplication; A (M, K), B (K, N); basic matmul",
```

Comment:
Consider adding two more TEST_CASEs (to the same sample or separate samples, up to you) that shows the ability to 1) broadcast and 2) transpose non-batch dims to serve as user documentation.